### ISSUE TITLE
[Bug]: Snapping do not works on vertexes (works on segments)

### ISSUE THREAD

**Issue by gioman** [3liz/lizmap-web-client#3693]
### What is the bug?

Snapping do not works on vertexes (works on segments)

### Steps to reproduce the issue

Configure your project with a PostGIS layer and set it as editable in LM plugin, vertexes only. > not working

Se snappjng to segments > work (but still no snapping on vertexes)

### Versions

LMWC 3.6.3 Latest version of LM plugin and LM server plugin. 

### Check Lizmap plugin

- [X] I have done the step just before in the Lizmap QGIS desktop plugin before opening this ticket. Otherwise, my ticket is not considered valid and might get closed.

### QGIS server version, only if the section above doesn't mention the QGIS Server version

3.28

### Operating system

Ubuntu 22.04, desktop and server

### Browsers

Firefox, Chrome

### Browsers version

Latest

**Comment by gioman** [comment]
Can anyone confirm?

**Comment by gioman** [comment]
Bumping my tickets before they go "quite",

**Comment by josemvm** [comment]
hi @gioman on LWC 3.6.5 everything works fine here

**Comment by josemvm** [comment]
qgis server and desktop 3.22.16

**Comment by gioman** [comment]
> qgis server and desktop 3.22.16

@josemvm thanks for the feedback. I'm on 3.28 and cannot downgrade to test.

**Comment by gioman** [comment]
I tried on a different machine still with QGIS Desktop and Server 3.28(.11) and with LMWC 3.6.6 and still can't see snapping to vertexes work. I tried on several layers, some complex. some very simple and yet I can only see snap on segments work.

**Comment by Gustry** [comment]
What kind of geometry are you editing ?
What kind of geometry are you snapping to ?

Which checkbox did you use for snapping ? Did you use "Intersection" ?

**Comment by gioman** [comment]
@Gustry 

> What kind of geometry are you editing ? 

polygons


> What kind of geometry are you snapping to ?

I'm trying to snap to the geometries of the layer I'm editing, not the ones of other layers 

> Which checkbox did you use for snapping ? 

only vertices > does not work
check also segments > works on segments, still not on vertices


>Did you use "Intersection" ?

no

**Comment by Gustry** [comment]
In PR #3929, it works well with the checkbox "Intersection".
I'm not able to check exactly what this checkbox is doing in LWC source code. But it's worth to have a look into label, tooltip etc. (`Snapping on inner vertices` like in QGIS ?)

**Comment by gioman** [comment]
> In PR #3929, it works well with the checkbox "Intersection".

@Gustry confirm, it works only also when "on intersection" checkbox is active. Is this expected? Sorry for asking, I hate wasting your time.

**Comment by gioman** [comment]
> For me it works for both "Segments" and "Intersections"

@Gustry yes it does, but on QGIS Desktop of "on intersection" is to snap on a point (without a vertex) that is defined by two crossing features, so is not used to snap to vertices.

**Comment by Gustry** [comment]
Yes, that's what I mean 1h ago, it's worth checking the label and tooltip in the plugin if that's the case (snapping on inner vertices).

**Comment by gioman** [comment]
> Yes, that's what I mean 1h ago, it's worth checking the label and tooltip in the plugin if that's the case (snapping on inner vertices).

@Gustry I'm lost but I don't want to argue (means > I understand what the tooltip says, but that should not be necessary to just snap on vertices, but it is ok to me if it works with "intersection" enabled).
